provider/resource_keycloak_identity_provider_token_exchange_scope_permission.go

```diff
@@ -0,0 +1,292 @@
+	}
+}
+
+func createClientPolicy(keycloakClient *keycloak.KeycloakClient, realmId, realmManagementClientId, providerAlias string, clients []string) (string, error) {
+	openidClientAuthorizationClientPolicy := &keycloak.OpenidClientAuthorizationClientPolicy{
+		RealmId:          realmId,
+		ResourceServerId: realmManagementClientId,
+		Name:             realmId + "_" + providerAlias + "_client_policy",
+		DecisionStrategy: "UNANIMOUS",
+		Logic:            "POSITIVE",
+		Type:             "client",
+		Clients:          clients,
+	}
+	err := keycloakClient.NewOpenidClientAuthorizationClientPolicy(openidClientAuthorizationClientPolicy)
+	if err != nil {
+		if strings.Contains(err.Error(), "409 Conflict") {
+			index := 1
+			for index <= 10 && err != nil {
+				openidClientAuthorizationClientPolicy.Name = providerAlias + "_" + strconv.Itoa(index) + "_client_policy"
+				err = keycloakClient.NewOpenidClientAuthorizationClientPolicy(openidClientAuthorizationClientPolicy)
+				index = index + 1
+			}
+		}
+	}
+	if err != nil {
+		return "", err
+	}
+	return openidClientAuthorizationClientPolicy.Id, nil
+}
+
+func unsetIdentityProviderTokenExchangeScopePermissionPolicy(keycloakClient *keycloak.KeycloakClient, realmId, providerAlias, policyId string) error {
+	identityProviderPermissions, err := keycloakClient.GetIdentityProviderPermissions(realmId, providerAlias)
+	if err != nil {
+		return err
+	}
+
+	realmManagementClient, err := keycloakClient.GetOpenidClientByClientId(realmId, "realm-management")
+	if err != nil {
+		return err
+	}
+
+	tokenExchangeScopedPermissionId, err := identityProviderPermissions.GetTokenExchangeScopedPermissionId()
+	if err != nil {
+		return err
+	}
+
+	permission, err := keycloakClient.GetOpenidClientAuthorizationPermission(realmId, realmManagementClient.Id, tokenExchangeScopedPermissionId)
+	if err != nil {
+		return err
+	}
+
+	permission.Policies = []string{}
+	err = keycloakClient.UpdateOpenidClientAuthorizationPermission(permission)
+	if err != nil {
+		return err
+	}
+
+	err = keycloakClient.DisableIdentityProviderPermissions(realmId, providerAlias)
+	if err != nil {
+		return err
+	}
+
+	_ = keycloakClient.DeleteOpenidClientAuthorizationClientPolicy(realmId, realmManagementClient.Id, policyId)
+
+	return nil
+}
+
+func resourceKeycloakIdentityProviderTokenExchangeScopePermissionCreate(data *schema.ResourceData, meta interface{}) error {
+	return resourceKeycloakIdentityProviderTokenExchangeScopePermissionUpdate(data, meta)
+}
+
+func resourceKeycloakIdentityProviderTokenExchangeScopePermissionUpdate(data *schema.ResourceData, meta interface{}) error {
+	keycloakClient := meta.(*keycloak.KeycloakClient)
+	realmId := data.Get("realm_id").(string)
+	providerAlias := data.Get("provider_alias").(string)
+	policyType := data.Get("policy_type").(string)
+	var clients []string
+
+	if v, ok := data.GetOk("clients"); ok {
```

Comment:
the `clients` attribute is required, so I don't think we need to use `data.GetOk` here

Reply:
true, But the idea Is that this later extended for different type of policies, at that moment this will be come optional attribute. It is only required when policy_type 'client' is selected